google-cloud-spanner/src/main/java/com/google/cloud/spanner/admin/database/v1/DatabaseAdminSettings.java

```diff
@@ -314,6 +318,20 @@ public static ApiClientHeaderProvider.Builder defaultApiClientHeaderProviderBuil
 
   /** Returns a new gRPC builder for this class. */
   public static Builder newBuilder() {
+    String hostAndPort = System.getenv(SPANNER_EMULATOR_HOST_ENV_VAR);
```

Comment:
@olavloite has suggested not to add logic on autogenerated files since there is a chance of this file getting overwritten during the next autogenerated PRs.
In NodeJS, we are writing a function similar to Python which can return a new autogenerated admin client with the emulator host set.

Reply:
yes, there's similar requirement in other client libs. 
Bigtable https://github.com/googleapis/java-bigtable/blob/main/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableTableAdminSettings.java#L56 has logic to use emulator in their classes.

Similar for Bigquery:  https://github.com/googleapis/java-bigtable/blob/main/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableTableAdminSettings.java#L146-L158